In [2]:
import pandas as pd

path_to_file = 'data.csv'
df = pd.read_csv(path_to_file)
df.head(5)

EXT   TYP                                                STR  \
0  M2   EXC  (SCACT in ('EE','SE','FP', 'EH', 'EL', 'EI', '...   
1  M2  EXCC  (SCACT in ('EE','SE', 'EH', 'EL', 'EF', 'CF', ...   
2  M2   INV  (SCACT in ('EE','SE','EH','GZ','MH') or SCACT=...   
3  M2   ISP  (SCACT in ('EE','SE','EH','GZ','MH','ZH','GL')...   
4  M2   MOB  (SCACT in ('EE','SE','EH','GZ','MH','ZH','GL')...   

                             SRT  SHR  
0                            NaN    Y  
1                            NaN    Y  
2                            NaN  NaN  
3                            NaN    Y  
4  #ACT: GZ, EE, SE, EH, EG, MH;    Y

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 147 entries, 0 to 146
Data columns (total 5 columns):
EXT    147 non-null object
TYP    147 non-null object
STR    147 non-null object
SRT    84 non-null object
SHR    56 non-null object
dtypes: object(5)
memory usage: 5.8+ KB


In [11]:
f = open('result.yml', 'w')
f.write('filter:\n filter:\n')

17

In [27]:
def filter_generator(df_ext, df_typ):
    for i in range(len(df)):
        if df['EXT'][i] == df_ext and df['TYP'][i] == df_typ:
            df_str = df['STR'][i]
            df_srt = df['SRT'][i]
            df_shr = df['SHR'][i]
            
            yml_shared = False
            if df_shr == 'Y':
                yml_shared = True
            
            
            print(str(df_srt) + ' \n\n' + str(df_str) + ' \n\n' + str(df_shr))

In [35]:
filter_generator('M2', 'EXC')

nan 

(SCACT in ('EE','SE','FP', 'EH', 'EL', 'EI', 'EF', 'ES', 'GZ', 'EJ', 'EK', 'AR','ER','E9','ET','VV','FU','SA','MH', 'AX','FY','A1','A2','A3','A4','A5') or SCACT='GO' and CONCAT(CONCAT(SCAB,SCAN),SCAS) IN (select CONCAT(CONCAT(CM9AB,CM9AN),CM9AS) from cm9pf)) and scai12='N' and scai14='N' and scai17='N' and scai20='N' and scai30='N' and scai82='N' and scp3r<>'998' and scai87='N' and SCAIC4='N' 

Y


(
    SCACT in 
        ('EE','SE','FP', 'EH', 'EL', 'EI', 'EF', 'ES', 'GZ', 'EJ', 'EK', 'AR','ER','E9','ET','VV','FU','SA','MH', 'AX','FY','A1','A2','A3','A4','A5')
        
    or SCACT='GO' 
    and CONCAT(CONCAT(SCAB,SCAN),SCAS) IN (select CONCAT(CONCAT(CM9AB,CM9AN),CM9AS) from cm9pf)) 
    and scai12='N' and scai14='N' and scai17='N' and scai20='N' and scai30='N' and scai82='N' and scp3r<>'998' and scai87='N' and SCAIC4='N' 
)

(
    SCACT in 
        ('EE','SE','FP', 'EH', 'EL', 'EI', 'EF', 'ES', 'GZ', 'EJ', 'EK','AR','ER','E9','ET','VV','FU','SA','MH','AX','ME','C2','FY','A1','A2','A3','A4','A5','ZH','GL') 
    
    or SCACT='GO' 
    and CONCAT(CONCAT(SCAB,SCAN),SCAS) IN (select CONCAT(CONCAT(CM9AB,CM9AN),CM9AS)from cm9pf)) 
    and scai12 ='N' and scai14 ='N' and scai17 ='N' and scai20 ='N' and scai30 ='N' and scai82 ='N' and scp3r<>'998' and scai87 ='N' and SCAIC4 ='N' 
    
    or (SCACT='R9' and scai14 ='N'and scai20 ='N' and SCAI30 = 'N'and scai80 = 'N' and SCP3R<> '998')) 

In [36]:
filter_generator('M2', 'OWN')

#CCY: RUR, EUR, USD, JPY;                                                  #AMA: DESCEND 

((SCACT in ('EE','SE','FP', 'EH', 'EL', 'EI', 'EF', 'ES', 'GZ', 'EJ', 'EK','AR','ER','E9','ET','VV','FU','SA','MH', 'AX','ME','C2','FY','A1','A2','A3','A4','A5','ZH','GL') or SCACT='GO' and CONCAT(CONCAT(SCAB,SCAN),SCAS) IN (select CONCAT(CONCAT(CM9AB,CM9AN),CM9AS)from cm9pf)) and scai12 ='N' and scai14 ='N' and scai17 ='N' and scai20 ='N' and scai30 ='N' and scai82 ='N' and scp3r<>'998' and scai87 ='N' and SCAIC4 ='N' or (SCACT='R9' and scai14 ='N'and scai20 ='N' and SCAI30 = 'N'and scai80 = 'N' and SCP3R<> '998')) 

Y


@helper.typeIn(account, 'R9')      SCACT='R9'

@helper.isValidTariffPlan(account) SCP3R<> '998'

    M2OWN:
      types: >
        EE, SE, FP, EH, EL, EI, EF, ES, GZ, EJ, EK, AR, ER, E9, ET,
        VV, FU, SA, MH, AX, ME, C2, FY, A1, A2, A3, A4, A5, ZH, GL,
        GO, R9
      expression: >
        (
          (@helper.typeIn(account, "GO") and goal != null)
          or @helper.typeNotIn(account, {"GO", "R9"})
        )
        and @helper.isValidTariffPlan(account)
        and account.ai14 == 'N' and account.ai17 == 'N'
        and account.ai20 == 'N' and account.ai30 == 'N'
        and account.ai12 == 'N' and account.ai82 == 'N' and account.ai87 == 'N'
        and account.aic4 == 'N'
        or (
          @helper.typeIn(account, 'R9') and @helper.isValidTariffPlan(account)
          and account.ai14 == 'N' and account.ai20 == 'N' and account.ai30 == 'N' and account.ai80 == 'N'
        )
      sorting: >
        @helper.sortBy({
          @helper.sortByCurrency({"RUR", "EUR", "USD", "JPY"}),
          @helper.sortByAmount("desc")
        })